extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -132,4 +182,30 @@ export class ScrollManager {
       this.onScroll_.bind(this)
     );
   }
+
+  /**
+   * Gets the cached layout rectangle of the root element
+   * @return {!Promise<!../../../src/layout-rect.LayoutRectDef>}
+   */
+  getInitRootElementRect_() {
+    if (!this.initialRootRectPromise_) {
+      return this.measureRootElement_();
+    }
+    return devAssert(this.initialRootRectPromise_);
+  }
+
+  /**
+   * Gets the layout rectangle of the root element
+   * @return {!Promise<!../../../src/layout-rect.LayoutRectDef>}
+   */
+  measureRootElement_() {
+    return this.mutator_.measureElement(() => {
+      const rectPromise = Promise.resolve(
```

Comment:
Why aren't you using `getClientRectAsync` directly?

Also return the promise immediately:

```js
// Either this or `getClientRectASync`
const rectPromise = return this.mutator_.measureElement(() =>
  this.viewport_.getLayoutRect(this.root_)
);
this.initialRootRectPromise_ =
  this.initialRootRectPromise_ || rectPromise;
return rectPromise;
```

Reply:
Discussed offline why `clientRect` doesn't make sense here, implemented the change